pkg/executor/sortexec/sort_partition.go

```diff
@@ -0,0 +1,310 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sortexec
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/failpoint"
+	"github.com/pingcap/tidb/pkg/types"
+	"github.com/pingcap/tidb/pkg/util/chunk"
+	"github.com/pingcap/tidb/pkg/util/disk"
+	"github.com/pingcap/tidb/pkg/util/memory"
+)
+
+var errSpillEmptyChunk = errors.New("can not spill empty chunk to disk")
+var errFailToAddChunk = errors.New("fail to add chunk")
+
+// It should be const, but we need to modify it for test.
+var spillChunkSize int = 4096
```

Comment:
Why not use e.MaxChunkSize()? 1024 is not enough?

Reply:
> Why not use e.MaxChunkSize()? 1024 is not enough?

I think the larger chunk size is the less io we will trigger.